Amqp/src/Akka.Streams.Amqp/AmqpRpcFlowStage.cs

```diff
@@ -158,57 +188,75 @@ private void HandleDelivery(IncomingMessage message)
-            }
+
+            public override void OnFailure(Exception ex) => _promise.TrySetException(ex);
+
             private class DefaultConsumer : DefaultBasicConsumer
             {
-                private readonly Action<IncomingMessage> _consumerCallback;
+                private readonly Action<CommittableIncomingMessage> _consumerCallback;
+                private readonly Action<ICommitCallback> _commitCallback;
                 private readonly Action<(string consumerTag, ShutdownEventArgs args)> _shutdownCallback;
-                public DefaultConsumer(Action<IncomingMessage> consumerCallback, Action<(string consumerTag, ShutdownEventArgs args)> shutdownCallback)
+
+                public DefaultConsumer(Action<CommittableIncomingMessage> consumerCallback, Action<ICommitCallback> commitCallback,
+                    Action<(string consumerTag, ShutdownEventArgs args)> shutdownCallback)
                 {
                     _consumerCallback = consumerCallback;
+                    _commitCallback = commitCallback;
                     _shutdownCallback = shutdownCallback;
                 }
+
                 public override void HandleBasicDeliver(string consumerTag, ulong deliveryTag, bool redelivered, string exchange, string routingKey,
                     IBasicProperties properties, byte[] body)
                 {
                     var envelope = Envelope.Create(deliveryTag, redelivered, exchange, routingKey);
-                    var incomingMessage = IncomingMessage.Create(ByteString.CopyFrom(body), envelope, properties);
-                    _consumerCallback?.Invoke(incomingMessage);
+                    var message = IncomingMessage.Create(ByteString.CopyFrom(body), envelope, properties);
+                    var committableMessage =
+                        new CommittableIncomingMessage(
+                            message,
+                            ack: multiple =>
+                            {
+                                var promise = new TaskCompletionSource<Done>();
+                                _commitCallback(new AckArguments(message.Envelope.DeliveryTag, multiple, promise));
+                                return promise;
+                            },
+                            nack: (multiple, requeue) =>
+                            {
+                                var promise = new TaskCompletionSource<Done>();
+                                _commitCallback(new NackArguments(message.Envelope.DeliveryTag, multiple, requeue, promise));
+                                return promise;
+                            });
+
+                    _consumerCallback?.Invoke(committableMessage);
                 }
+
                 public override void HandleBasicCancel(string consumerTag)
                 {
                     // non consumer initiated cancel, for example happens when the queue has been deleted.
                     _shutdownCallback?.Invoke((consumerTag, null));
```

Comment:
From what I see, this will never be null. No need for elvis operator.

Reply:
Done.